internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -413,7 +413,22 @@ func setSchedulingPolicy(p *processRunner, pid int, schedulingPolicy uint) error
 	return nil
 }
 
+//nolint:gocyclo
 func (p *processRunner) run(eventSink events.Recorder) error {
+	if cgroups.Mode() == cgroups.Unified {
+		cg, err := cgroup.CreateCgroupV2(p.opts.CgroupPath)
+		if err != nil {
+			return fmt.Errorf("error creating cgroup: %w", err)
+		}
+
+		defer func() {
+			err := cg.Delete()
+			if err != nil {
```

Comment:
no need for the os.IsNotExist error here?

Reply:
Yes, in case of containerd the containerd will have removed the cgroup on container exit (but not if we haven't spawned the container and exited early due to an error). I cannot see in what situation might the cgroup be cleaned up in case of a directly launched process